src/test/java/io/managed/services/test/billing/BillingMetricsTest.java

```diff
@@ -1,22 +1,326 @@
+ * </ul>
+ */
 @Log4j2
 public class BillingMetricsTest extends TestBase {
-    private ObservatoriumClient client;
+
+    // class uses long live kafka instance, as waiting for metrics to be available and stable in newly created instance can take too long.
+    public static final String KAFKA_INSTANCE_NAME = "mk-e2e-ll-" + Environment.LAUNCH_KEY;
+    public static final String SERVICE_ACCOUNT_NAME = "mk-billing-sa-" + Environment.LAUNCH_KEY;
+
+    public static final String TOPIC_NAME = "billing_test";
+
+    private PrometheusWebClient observatoriumClient;
+    private PrometheusWebClient prometheusWebClient;
+
+    private KafkaRequest kafka;
+    private ServiceAccount serviceAccount;
+
+    private KafkaMgmtApi kafkaMgmtApi;
+    private SecurityMgmtApi securityMgmtApi;
+    private KafkaInstanceApi kafkaInstanceApi;
+
+    private Map<String, PrometheusWebClient.Snapshot> metricToSnapshotMap = new HashMap<>();
+
+    private static final String METRIC_STORAGE = "kafka_broker_quota_totalstorageusedbytes";
+    private static final String METRIC_TRAFFIC_IN = "haproxy_server_bytes_in_total";
+    private static final String METRIC_TRAFFIC_OUT = "haproxy_server_bytes_out_total";
+    private static final String METRIC_CLUSTER_HOURS = "kafka_id:strimzi_resource_state:max_over_time1h";
+
+    // size and count of messages to be produced/ consumed
+    private final int messageSize = 1024 * 128;
+    private final int messageCount = 40;
+    // number of consumers consuming data
+    private final int consumerCount = 3;
 
     @BeforeClass
-    public void setup() throws ObservatoriumException {
-        client = new ObservatoriumClient();
+    @SneakyThrows
+    public void bootstrap() {
+        assertNotNull(Environment.PRIMARY_USERNAME, "the PRIMARY_USERNAME env is null");
+        assertNotNull(Environment.PRIMARY_PASSWORD, "the PRIMARY_PASSWORD env is null");
+        assertNotNull(Environment.PROMETHEUS_WEB_CLIENT_ACCESS_TOKEN, "the PROMETHEUS_WEB_CLIENT_ACCESS_TOKEN env is null");
+
+        var apps = ApplicationServicesApi.applicationServicesApi(
+                Environment.PRIMARY_USERNAME,
+                Environment.PRIMARY_PASSWORD);
+
+        this.kafkaMgmtApi = apps.kafkaMgmt();
+        this.securityMgmtApi = apps.securityMgmt();
+
+        // clients
+        // build observatorium client, this client points on locally running container/process
+        this.observatoriumClient = new PrometheusWebClientBuilder()
+                .withBaseUrl("http://localhost:8085")
+                .withUrlResourcePath("/api/metrics/v1/managedkafka/api/v1/query")
+                .build();
+        // build prometheus client with all necessary config, this clients points on server running on oc cluster
+        this.prometheusWebClient = new PrometheusWebClientBuilder()
+                .withBaseUrl("https://kafka-prometheus-managed-application-services-observability.apps.mk-stage-0622.bd59.p1.openshiftapps.com")
```

Comment:
Could this be obtained from an env var?

Reply:
i can see scenario where we will obtain this from env varable, or even where we will not obtain it at all, 
I left it as it is and would change it later (that is when we will have oc client for stage cluster in place,) In that scenario we could obtain something like `oc get route ....` and parse this url directly without need for additional env variable.  
User would later only provide name of route which would make code more prone to change and also easier to execute in environments different from stage.